theano/tensor/nnet/abstract_conv.py

```diff
@@ -1690,9 +1690,9 @@ def perform(self, node, inp, out_):
             img = new_img
         if not self.filter_flip:
             if self.unshared is True:
```

Comment:
It's certainly good to have it there, and hopefully it's somewhat temporary anyway. However, I do think I'd like to have an explicit check in `perform` as well. It's good to have the check and warning close to where the limitation actually is, in my opinion. It's useful as documentation and also makes it future-proof: if at some point the check is removed from BaseAbstractConv, it's better for `perform` to crash than to silently run a shared convolution.

Compare this with the 2D/3D limitation of the convolution: that's also explicitly checked in [`BaseAbstractConv.conv`](https://github.com/Theano/Theano/blob/master/theano/tensor/nnet/abstract_conv.py#L1540).